nuclear/tools/module.py

```diff
@@ -173,3 +183,72 @@ def generate_test(parts):
     return template.format()
 
 
+def module_unused(**kwargs):
+    roles_path = b.cmake_cache["NUCLEAR_ROLES_DIR"]
+    modules_path = b.cmake_cache["NUCLEAR_MODULE_DIR"]
+    existing_modules = {}
+    missing_modules = {}
+    commented_modules = {}
+    used_modules = []
+    unused_commented_modules = []
+
+    # a list of folders that could be within a module
+    skip_list = ['src', 'data', 'config', 'tests', 'report']
```

Comment:
Rather than doing it this way, just search for files named CMakeLists.txt that contain a call to NUClearModule(...)